packages/cli/src/commands/add-deps.ts

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright 2020 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { paths } from '../lib/paths';
+import { ESLint } from 'eslint';
+import { join as joinPath, basename } from 'path';
+import fs from 'fs-extra';
+import { isChildPath } from '@backstage/cli-common';
+import { PackageGraph } from '../lib/monorepo';
+
+function isTestPath(filePath: string) {
+  if (!isChildPath(joinPath(paths.targetDir, 'src'), filePath)) {
+    return true;
+  }
+  const name = basename(filePath);
+  return (
+    name.startsWith('setupTests.') ||
+    name.includes('.test.') ||
+    name.includes('.stories.')
+  );
+}
+
+export async function command() {
+  const pkgJsonPath = paths.resolveTarget('package.json');
+  const pkg = await fs.readJson(pkgJsonPath);
+  if (pkg.workspaces) {
+    throw new Error(
+      'Adding dependencies to the workspace root is not supported',
+    );
+  }
+
+  const packages = await PackageGraph.listTargetPackages();
+  const localPackageVersions = new Map(
+    packages.map(p => [p.packageJson.name, p.packageJson.version]),
+  );
+
+  const eslint = new ESLint({
+    cwd: paths.targetDir,
+    overrideConfig: {
+      plugins: ['monorepo'],
+      rules: {
+        'import/no-extraneous-dependencies': [
+          'error',
+          {
+            devDependencies: [
+              `!${joinPath(paths.targetDir, 'src/**')}`,
+              joinPath(paths.targetDir, 'src/**/*.test.*'),
+              joinPath(paths.targetDir, 'src/**/*.stories.*'),
+              joinPath(paths.targetDir, 'src/setupTests.*'),
+            ],
```

Comment:
Is there a case for adding things from the dev folder too where appropriate?

Reply:
First pattern takes care of that, anything outside src